apis/core.oam.dev/v1alpha2/environment_types.go

```diff
@@ -0,0 +1,71 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha2
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EnvironmentSpec defines vela workload of Environment
+type EnvironmentSpec struct {
+
+	// FIXME vela Namespace duplicate from k8s namesapce
+
+	// +kubebuilder:validation:Type=string
+	// +kubebuilder:validation:Required
+	Namespace string `json:"namespace"`
+
+	// +kubebuilder:validation:Type=string
+	// +optional
+	Email string `json:"email,omitempty"`
+
+	// +kubebuilder:validation:Type=string
+	// +optional
+	Domain string `json:"domain,omitempty"`
+
+	// +kubebuilder:validation:Type=string
+	// +optional
+	Issuer string `json:"issuer,omitempty"`
+
+	// +optional
+	CapabilityList []string `json:"capabilityList,omitempty"`
+}
+
+// EnvironmentStatus defines the observed state of Environment
+type EnvironmentStatus struct {
+	// TODO Support aggregation workload status?
+}
+
+// +kubebuilder:object:root=true
+
+// Environment is vela Managed environments
+// +kubebuilder:resource:scope=Cluster
```

Comment:
I think the CRD can be namespace scope

Reply:
I use a namespace scope, remove `spec.namespace`. it will be created in the user-specified `Namespace`.